### ISSUE TITLE
onNotification is not called when local notification arrives and app is in background in android

### ISSUE THREAD

**Issue by anuj47billion** [zo0r/react-native-push-notification#1601]
Code for Local Noification

PushNotification.localNotification({
      autoCancel: true,
      title: 'Test Title',
      message: 'Test Message',
      vibrate: true,
      vibration: 1000,
      playSound: false,
      data: {},
    });

Trying to get in configure onNotification

PushNotification.configure({
  onRegister: function(token) {
    console.log('TOKEN:', token);
  },
  onNotification: function(notification) {
    console.log('NOTIFICATION:', notification);
  },
  permissions: {
    alert: true,
    badge: true,
    sound: true,
  },
  popInitialNotification: true,
  requestPermissions: true,
});

**Comment by Dallas62** [comment]
Hi,

`onNotification` will not be called for local/scheduled notifications if there is no user interaction.

**Comment by anuj47billion** [comment]
@Dallas62 , It still not working when the app is in background

**Comment by Dallas62** [comment]
Does the user interact with the notification ?

**Comment by anuj47billion** [comment]
Yes set the userInteraction parameter as true

**Comment by Dallas62** [comment]
So the user Press the notification and `onNotification` is not called ?

Does `.configure` inside a component ?
Do you have a splashscreen in your application ?

**Comment by anuj47billion** [comment]
Yes onNotification in not called when user press in notifcation when app is in background.
.configure is not inside a component.
Not having splashscreen

**Comment by Dallas62** [comment]
If everything is like you described, this is like my first comment:
`onNotification` will not be called for local/scheduled notifications if there is no user interaction (user pressing notification).

In others words: this is the expected behavior you described

**Comment by Dallas62** [comment]
Did you test the example project ?

**Comment by anuj47billion** [comment]
I compared AndroidManifest.xml from example folder and added all changes

**Comment by Dallas62** [comment]
Did you looked at: https://github.com/zo0r/react-native-push-notification/issues/1592

**Comment by anuj47billion** [comment]
Yes @Dallas62 but not succeed

**Comment by Dallas62** [comment]
Can't do more without a reproducible example.

**Comment by anuj47billion** [comment]
@Dallas62 , I also used PushNotification.popInitialNotification to fulfill my requirement but it also returns undefined randomly

**Comment by Dallas62** [comment]
The initial notification is the notification which start the application, bring back to foreground an application in background doesn't change or set the initial notification.

**Comment by kirantripathi** [comment]
<img width="510" alt="Screen Shot 2020-09-02 at 15 00 59" src="https://user-images.githubusercontent.com/17666887/91963075-1f243800-ed2d-11ea-8ba1-46f5d861202b.png">
onNotification function gets called in android , but. in ios  I am having issue to get response in onNotification function . Do. you have any suggestion on that @Dallas62

**Comment by Dallas62** [comment]
Yes this is an issue from the iOS library, looking into it.
https://github.com/react-native-community/push-notification-ios/pull/122

**Comment by kirantripathi** [comment]
> Yes this is an issue from the iOS library, looking into it.
> [react-native-community/push-notification-ios#122](https://github.com/react-native-community/push-notification-ios/pull/122)

do we have any idea on how we can tackle it ?

**Comment by Dallas62** [comment]
Hi!
You can test the version on `dev` branch, react-native-community/push-notification-ios#122 has been merged recently.
